src/main/java/redis/clients/jedis/JedisBroadcast.java

```diff
@@ -0,0 +1,130 @@
+package redis.clients.jedis;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import redis.clients.jedis.Protocol.Command;
+import redis.clients.jedis.Protocol.Keyword;
+import redis.clients.jedis.args.FlushMode;
+import redis.clients.jedis.providers.ConnectionProvider;
+import redis.clients.jedis.search.FTCreateParams;
+import redis.clients.jedis.search.IndexOptions;
+import redis.clients.jedis.search.Schema;
+import redis.clients.jedis.search.SearchProtocol.SearchCommand;
+import redis.clients.jedis.search.SearchProtocol.SearchKeyword;
+import redis.clients.jedis.search.schemafields.SchemaField;
+import redis.clients.jedis.util.Pool;
+
+public class JedisBroadcast {
+
+  private final ConnectionProvider provider;
+
+  public JedisBroadcast(UnifiedJedis jedis) {
+    this(jedis.provider);
+  }
+
+  public JedisBroadcast(ConnectionProvider provider) {
+    if (provider == null) {
+      throw new NullPointerException("ConnectionProvider is null.");
+    }
+    this.provider = provider;
+  }
+
+  public final <T> Map<?, BroadcastResponse<T>> broadcastCommand(CommandObject<T> commandObject) {
+    Map<?, ?> connectionMap = provider.getConnectionMap();
+    Map<Object, BroadcastResponse<T>> responseMap = new HashMap<>(connectionMap.size(), 1f);
+    for (Map.Entry<? extends Object, ? extends Object> entry : connectionMap.entrySet()) {
+      Object key = entry.getKey();
+      Object connection = entry.getValue();
+      try {
+        responseMap.put(key, new BroadcastResponse<>(executeCommand(connection, commandObject)));
+      } catch (RuntimeException re) {
+        responseMap.put(key, new BroadcastResponse<>(re));
+      }
+    }
+    return responseMap;
+  }
+
+  private <T> T executeCommand(Object connection, CommandObject<T> commandObject) {
+    if (connection instanceof Connection) {
+      return ((Connection) connection).executeCommand(commandObject);
+    }
+    if (connection instanceof Pool) {
+      try (Connection _conn = ((Pool<Connection>) connection).getResource()) {
+        return _conn.executeCommand(commandObject);
+      }
+    }
+    throw new IllegalStateException(connection.getClass() + "is not supported.");
```

Comment:
Any reason to not:

if (connection instanceof Connection)
else if (connection instanceof Pool)
// throw here

Reducing multiple ifs to a single if else one makes thing function cheaper.